### ISSUE TITLE
ERROR [io.ver.cor.htt.imp.HttpClientRequestImpl] (vert.x-eventloop-thread-1) {} Connection was closed

### ISSUE THREAD

**Issue by rickbear** [quarkusio/quarkus#39695]
### Describe the bug

I have found log entries in my log with the following message:
`ERROR [io.ver.cor.htt.imp.HttpClientRequestImpl] (vert.x-eventloop-thread-1) {} Connection was closed`
When I try to trace the error in other logs I can see no other errors. It seems like everything behaves like it should. Though I still wonder if the above message could mean something I should be worried about?

[This bug](https://github.com/quarkusio/quarkus/issues/38335) seems to be very similar to my problem on a ConnectionBase instead of HttpClientRequestImpl.

I am using Quarkus 3.8.1 with these dependencies:
```
io.quarkus:quarkus-smallrye-openapi
io.quarkus:quarkus-rest-client-reactive-jackson
io.quarkus:quarkus-resteasy-reactive-jackson
io.quarkus:quarkus-smallrye-fault-tolerance
io.quarkus:quarkus-arc
io.quarkus:quarkus-smallrye-context-propagation
io.quarkus:quarkus-smallrye-health
io.quarkus:quarkus-opentelemetry
io.opentelemetry:opentelemetry-extension-trace-propagators
io.quarkus:quarkus-smallrye-jwt
io.quarkus:quarkus-cache
```

### Expected behavior

That either there would be no error log or there would be a more descriptive error log which points me in the correct direction.

### Actual behavior

This log entry is shown:
`ERROR [io.ver.cor.htt.imp.HttpClientRequestImpl] (vert.x-eventloop-thread-1) {} Connection was closed`
for like it seems no particular reason.

### How to Reproduce?

It doesn't seem there is a special trigger for the log entry. There are hundreds of calls per minute to this service. So there should be either a lot of entries or none. It appears randomly it seems.

### Output of `uname -a` or `ver`

GNU/Linux

### Output of `java -version`

Java 17

### Quarkus version or git rev

Quarkus 3.8.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.6

### Additional information

I know uname -a and java -version is not very specific. The code is build and deployed on a system I have no much control of. I can find more specific informations about it if it is of any importance for the issue but it might take a while because of the easter holidays.

**Comment by Leokao4Autopass** [comment]
I have the same issue.  the Error message happened in my production environment, but I can find any exception stack trace in log, this so hard identify the root cause.

the log snippet
```
2024-06-18 17:17:16,047 INFO  [com.pkl.aut.mq.goo.pub.GoogleDomainEventBus$postAll$1$1:40] [] [] published Successful. message id: 11537991230112863.
2024-06-18 17:17:18,981 ERROR [io.ver.cor.htt.imp.HttpClientRequestImpl:-1] [] [] Connection was closed
2024-06-18 17:17:35,415 INFO  [com.pkl.aut.com.uti.RedissonLockFactory:17] [f791bfddf6f7961d4817dddc2d8f50d6] [false] Redisson tryLock by key: lock:ISSUE_INVOICE_2292672
```

**Comment by jponge** [comment]
Closing due to a lack of reproducer.

**Comment by dcdh** [comment]
I've got the same error because due to this kind of code (pseudo code here)

```java
    @Inject
    ManagedExecutor executor;

List<CompletableFuture<...,...>>> futures = new ArrayList<>();

for (int i = 0; i < 1000; i++) {
     futures.add(executor.supplyAsync(() -> {
          // do a rest client remote call
     });
}
CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
```

I guess, I get the issue because due to the pool size defined to 50 by default : [quarkus.rest-client.connection-pool-size](https://quarkus.io/guides/rest-client#quarkus-rest-client-config_quarkus-rest-client-connection-pool-size) and by default all CompletableFuture will be executed in parallel.

To avoid it, the fix was to define the number of async completion stage execution this way

```java
    @Inject
    @ManagedExecutorConfig(maxAsync = 30)
    ManagedExecutor executor;
```

The maxAsync is bellow the connection pool size of the rest client connection.

From @jponge comment, we miss a reproducer so I do not know if it will help or not but in my case I was able to fix my issue this way.
